feat(prebuiltconfig/neo4j): Add prebuiltconfig support for neo4j (#1352)
googleapis/genai-toolbox: docs/en/how-to/connect-ide/neo4j_mcp.md

Reviewer: Can you add this path to .github/release-please.yml?

Author: Added

---
title: Neo4j using MCP
type: docs
weight: 2
description: "Connect your IDE to Neo4j using Toolbox."
---

[Model Context Protocol (MCP)](https://modelcontextprotocol.io/introduction) is an open protocol for connecting Large Language Models (LLMs) to data sources like Neo4j. This guide covers how to use [MCP Toolbox for Databases][toolbox] to expose your developer assistant tools to a Neo4j instance:

* [Cursor][cursor]
* [Windsurf][windsurf] (Codium)
* [Visual Studio Code][vscode] (Copilot)
* [Cline][cline] (VS Code extension)
* [Claude desktop][claudedesktop]
* [Claude code][claudecode]
* [Gemini CLI][geminicli]
* [Gemini Code Assist][geminicodeassist]

[toolbox]: https://github.com/googleapis/genai-toolbox
[cursor]: #configure-your-mcp-client
[windsurf]: #configure-your-mcp-client
[vscode]: #configure-your-mcp-client
[cline]: #configure-your-mcp-client
[claudedesktop]: #configure-your-mcp-client
[claudecode]: #configure-your-mcp-client
[geminicli]: #configure-your-mcp-client
[geminicodeassist]: #configure-your-mcp-client

## Set up the database

1.  [Create or select a Neo4j instance.](https://neo4j.com/cloud/platform/aura-graph-database/)

## Install MCP Toolbox

> 1.  Download the latest version of Toolbox as a binary. Select the [correct binary](https://github.com/googleapis/genai-toolbox/releases) corresponding to your OS and CPU architecture. You are required to use Toolbox version V0.10.0+:

Reviewer: Can you update this version and the others in the doc to v0.15.0?

* [Cursor][cursor]
* [Windsurf][windsurf] (Codium)
* [Visual Studio Code][vscode] (Copilot)
* [Cline][cline] (VS Code extension)
* [Claude desktop][claudedesktop]
* [Claude code][claudecode]
* [Gemini CLI][geminicli]
* [Gemini Code Assist][geminicodeassist]

[toolbox]: https://github.com/googleapis/genai-toolbox
[cursor]: #configure-your-mcp-client
[windsurf]: #configure-your-mcp-client
[vscode]: #configure-your-mcp-client
[cline]: #configure-your-mcp-client
[claudedesktop]: #configure-your-mcp-client
[claudecode]: #configure-your-mcp-client
[geminicli]: #configure-your-mcp-client
[geminicodeassist]: #configure-your-mcp-client

## Set up the database

1.  [Create or select a Neo4j instance.](https://neo4j.com/cloud/platform/aura-graph-database/)

## Install MCP Toolbox

1.  Download the latest version of Toolbox as a binary. Select the [correct binary](https://github.com/googleapis/genai-toolbox/releases) corresponding to your OS and CPU architecture. You are required to use Toolbox version v0.15.0+:

>     {{< tabpane persist=header >}}
>     {{< tab header="linux/amd64" lang="bash" >}}
> 
>     ```bash
>     curl -O [https://storage.googleapis.com/genai-toolbox/v0.15.0/linux/amd64/toolbox](https://storage.googleapis.com/genai-toolbox/v0.15.0/linux/amd64/toolbox)
>     ```
> 
>     {{< /tab >}}
>     {{< tab header="darwin/arm64" lang="bash" >}}
> 
>     ```bash
>     curl -O [https://storage.googleapis.com/genai-toolbox/v0.15.0/darwin/arm64/toolbox](https://storage.googleapis.com/genai-toolbox/v0.15.0/darwin/arm64/toolbox)
>     ```
> 
>     {{< /tab >}}
>     {{< tab header="darwin/amd64" lang="bash" >}}
> 
>     ```bash
>     curl -O [https://storage.googleapis.com/genai-toolbox/v0.15.0/darwin/amd64/toolbox](https://storage.googleapis.com/genai-toolbox/v0.15.0/darwin/amd64/toolbox)
>     ```
> 
>     {{< /tab >}}
>     {{< tab header="windows/amd64" lang="bash" >}}
> 
>     ```bash
>     curl -O [https://storage.googleapis.com/genai-toolbox/v0.15.0/windows/amd64/toolbox.exe](https://storage.googleapis.com/genai-toolbox/v0.15.0/windows/amd64/toolbox.exe)
>     ```
> 
>     {{< /tab >}}
>     {{< /tabpane >}}
>     <!-- {x-release-please-end} -->

Reviewer 2: ```suggestion
   <!-- {x-release-please-start-version} -->
   {{< tabpane persist=header >}}
{{< tab header="linux/amd64" lang="bash" >}}
curl -O https://storage.googleapis.com/genai-toolbox/v0.14.0/linux/amd64/toolbox
{{< /tab >}}

{{< tab header="darwin/arm64" lang="bash" >}}
curl -O https://storage.googleapis.com/genai-toolbox/v0.14.0/darwin/arm64/toolbox
{{< /tab >}}

{{< tab header="darwin/amd64" lang="bash" >}}
curl -O https://storage.googleapis.com/genai-toolbox/v0.14.0/darwin/amd64/toolbox
{{< /tab >}}

{{< tab header="windows/amd64" lang="bash" >}}
curl -O https://storage.googleapis.com/genai-toolbox/v0.14.0/windows/amd64/toolbox.exe
{{< /tab >}}
{{< /tabpane >}}
    <!-- {x-release-please-end} -->
```

1. Make the binary executable:

    ```bash
    chmod +x toolbox
    ```

1. Verify the installation:

    ```bash
    ./toolbox --version
    ```

## Configure your MCP Client

{{< tabpane text=true >}}
{{% tab header="Claude code" lang="en" %}}

1.  Install [Claude Code](https://docs.anthropic.com/en/docs/agents-and-tools/claude-code/overview).

> 2.  Create a `.mcp.json` file in your project root if it doesn't exist.
> 3.  Add the following configuration, replace the environment variables with your values, and save:

Reviewer 2: ```suggestion
1.  Create a `.mcp.json` file in your project root if it doesn't exist.
1.  Add the following configuration, replace the environment variables with your values, and save:
```

It will automatically number it accordingly. This is preferred incase a line was removed (or added) in the future, we don't have to manually update the numbers.

Please also update the remaining tabs.

    ```json
    {
      "mcpServers": {
        "neo4j": {
          "command": "./PATH/TO/toolbox",
          "args": ["--prebuilt","neo4j","--stdio"],
          "env": {
            "NEO4J_URI": "",
            "NEO4J_DATABASE": "",
            "NEO4J_USERNAME": "",
            "NEO4J_PASSWORD": ""
          }
        }
      }
    }
    ```

> 4.  Restart Claude code to apply the new configuration.

Reviewer 2: ```suggestion
1.  Restart Claude code to apply the new configuration.
```

{{% /tab %}}
{{% tab header="Claude desktop" lang="en" %}}

> 1.  Open [Claude desktop](https://claude.ai/download) and navigate to Settings.
> 2.  Under the Developer tab, tap Edit Config to open the configuration file.
> 3.  Add the following configuration, replace the environment variables with your values, and save:
> 
>     ```json
>     {
>       "mcpServers": {
>         "neo4j": {
>           "command": "./PATH/TO/toolbox",
>           "args": ["--prebuilt","neo4j","--stdio"],
>           "env": {
>             "NEO4J_URI": "",
>             "NEO4J_DATABASE": "",
>             "NEO4J_USERNAME": "",
>             "NEO4J_PASSWORD": ""
>           }
>         }
>       }
>     }
>     ```
> 
> 4.  Restart Claude desktop.
> 5.  From the new chat screen, you should see a hammer (MCP) icon appear with the new MCP server available.

Reviewer 2: ```suggestion
1.  Open [Claude desktop](https://claude.ai/download) and navigate to Settings.
1.  Under the Developer tab, tap Edit Config to open the configuration file.
1.  Add the following configuration, replace the environment variables with your values, and save:

    ```json
    {
      "mcpServers": {
        "neo4j": {
          "command": "./PATH/TO/toolbox",
          "args": ["--prebuilt","neo4j","--stdio"],
          "env": {
            "NEO4J_URI": "",
            "NEO4J_DATABASE": "",
            "NEO4J_USERNAME": "",
            "NEO4J_PASSWORD": ""
          }
        }
      }
    }
    ```

1.  Restart Claude desktop.
1.  From the new chat screen, you should see a hammer (MCP) icon appear with the new MCP server available.
```

{{% /tab %}}
{{% tab header="Cline" lang="en" %}}

> 1.  Open the [Cline](https://github.com/cline/cline) extension in VS Code and tap the **MCP Servers** icon.
> 2.  Tap Configure MCP Servers to open the configuration file.
> 3.  Add the following configuration, replace the environment variables with your values, and save:
> 
>     ```json
>     {
>       "mcpServers": {
>         "neo4j": {
>           "command": "./PATH/TO/toolbox",
>           "args": ["--prebuilt","neo4j","--stdio"],
>           "env": {
>             "NEO4J_URI": "",
>             "NEO4J_DATABASE": "",
>             "NEO4J_USERNAME": "",
>             "NEO4J_PASSWORD": ""
>           }
>         }
>       }
>     }
>     ```
> 
> 4.  You should see a green active status after the server is successfully connected.

Reviewer 2: ```suggestion
1.  Open the [Cline](https://github.com/cline/cline) extension in VS Code and tap the **MCP Servers** icon.
1.  Tap Configure MCP Servers to open the configuration file.
1.  Add the following configuration, replace the environment variables with your values, and save:

    ```json
    {
      "mcpServers": {
        "neo4j": {
          "command": "./PATH/TO/toolbox",
          "args": ["--prebuilt","neo4j","--stdio"],
          "env": {
            "NEO4J_URI": "",
            "NEO4J_DATABASE": "",
            "NEO4J_USERNAME": "",
            "NEO4J_PASSWORD": ""
          }
        }
      }
    }
    ```

1.  You should see a green active status after the server is successfully connected.
```

            "NEO4J_DATABASE": "",
            "NEO4J_USERNAME": "",
            "NEO4J_PASSWORD": ""
          }
        }
      }
    }
    ```
{{% /tab %}}
{{% tab header="Gemini Code Assist" lang="en" %}}

1.  Install the [Gemini Code Assist](https://marketplace.visualstudio.com/items?itemName=Google.geminicodeassist) extension in Visual Studio Code.
2.  Enable Agent Mode in Gemini Code Assist chat.
3.  In your working directory, create a folder named `.gemini`. Within it, create a `settings.json` file.
4.  Add the following configuration, replace the environment variables with your values, and then save:

    ```json
    {
      "mcpServers": {
        "neo4j": {
          "command": "./PATH/TO/toolbox",
          "args": ["--prebuilt","neo4j","--stdio"],
          "env": {
            "NEO4J_URI": "",
            "NEO4J_DATABASE": "",
            "NEO4J_USERNAME": "",
            "NEO4J_PASSWORD": ""
          }
        }
      }
    }
    ```
{{% /tab %}}
{{< /tabpane >}}

## Use Tools

Your AI tool is now connected to Neo4j using MCP. Try asking your AI assistant to get the graph schema or execute Cypher statements.

The following tools are available to the LLM:

> 1.  **schema**: extracts the complete database schema, including details about node labels, relationships, properties, constraints, and indexes.

Reviewer 2: ```suggestion
1.  **get_schema**: extracts the complete database schema, including details about node labels, relationships, properties, constraints, and indexes.
```

> 2.  **execute_cypher**: executes any arbitrary Cypher statement.

Reviewer 2: ```suggestion
1.  **execute_cypher**: executes any arbitrary Cypher statement.
```
